VideoApi/VideoApi/Services/VirtualRoomService.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using VideoApi.DAL.Commands;
+using VideoApi.DAL.Commands.Core;
+using VideoApi.DAL.Queries;
+using VideoApi.DAL.Queries.Core;
+using VideoApi.Domain;
+using VideoApi.Domain.Enums;
+using VideoApi.Services.Contracts;
+using VideoApi.Services.Kinly;
+
+namespace VideoApi.Services
+{
+    public class VirtualRoomService : IVirtualRoomService
+    {
+        private readonly IKinlyApiClient _kinlyApiClient;
+        private readonly ILogger<VirtualRoomService> _logger;
+        private readonly ICommandHandler _commandHandler;
+        private readonly IQueryHandler _queryHandler;
+
+        public VirtualRoomService(IKinlyApiClient kinlyApiClient, ILogger<VirtualRoomService> logger,
+            ICommandHandler commandHandler, IQueryHandler queryHandler)
+        {
+            _kinlyApiClient = kinlyApiClient;
+            _logger = logger;
+            _commandHandler = commandHandler;
+            _queryHandler = queryHandler;
+        }
+
+        public async Task<Room> GetOrCreateAnInterpreterVirtualRoom(Conference conference, Participant participant)
+        {
+            var conferenceId = conference.Id;
+
+            // get a room
+            var roomPrefix = "Interpreter";
+            var query = new GetAvailableRoomByRoomTypeQuery(VirtualCourtRoomType.Civilian, conferenceId);
+            var listOfRooms = await _queryHandler.Handle<GetAvailableRoomByRoomTypeQuery, List<Room>>(query);
+            var interpreterRooms = listOfRooms.Where(x => x.Label.Contains(roomPrefix)).ToList();
+            var room = GetRoomForParticipant(conferenceId, interpreterRooms, participant) ??
+                       await CreateAnInterpreterRoom(conference, interpreterRooms.Count);
```

Comment:
This might not work as intended.
`GetAvailableRoomByRoomTypeQuery` only returns Live rooms that are rooms with participants, so the count may not represent the number, maybe we could do something like
`interpreterRooms.Select(x => int.TryParse(x.DisplayName.Replace(roomPrefix, string.Empty), out var n) ? n : 0)).Max();`

Reply:
thanks. just a tip
use `DefaultIfEmpty` to handle empty lists else Max would throw an exception